### ISSUE TITLE
String methods should have `this` parameters in `lib.d.ts`

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#9654]
``` ts
// @noImplicitThis: true
String.prototype.indexOf = function(substr, start) {
    this./**/
}
```

Expected: Method suggestions. No error.
Actual: Nothing. `this: any` error.

Other types in lib.d.ts should probably also have `this` added too.

**Comment by RyanCavanaugh** [comment]
This work automatically now
